Sources/NIOCore/EventLoop.swift

```diff
@@ -478,8 +481,20 @@ public struct NIODeadline: Equatable, Hashable, NIOSendable {
         self._uptimeNanoseconds = nanoseconds
     }
 
+
+    @inline(__always)
+    private static func timeNow() -> UInt64 {
+#if os(Linux)
+        var ts = timespec()
+        clock_gettime(CLOCK_MONOTONIC, &ts)
+        return UInt64(ts.tv_sec) &* 1_000_000_000 &+ UInt64(ts.tv_nsec)
```

Comment:
We need comments added to explain why it's safe for this math to be unchecked. In this case this boils down to: "UInt64.max nanoseconds is more than 580 years, and the odds that this code will still be running 530 years from now is very, very low, so as a practical matter this will never overflow."